app/scripts/modules/core/src/application/config/footer/ConfigSectionFooter.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+
+export interface IConfigSectionFooterProps {
+  isDirty: boolean;
+  isValid: boolean;
+  isSaving: boolean;
+  saveError: boolean;
+  onRevertClicked: () => void;
+  onSaveClicked: () => void;
+}
+
+@autoBindMethods
+export class ConfigSectionFooter extends React.Component<IConfigSectionFooterProps> {
+  public render() {
+    const { isValid, isDirty, isSaving, saveError, onRevertClicked, onSaveClicked } = this.props;
+
+    if (!isDirty) {
+      return (
+        <div className="row footer">
+          <div className="col-md-3" />
+          <div className="col-md-9 text-right">
+            <span className="btn btn-link disabled"><span className="fa fa-check-circle-o"/> In sync with server</span>
+          </div>
+        </div>
+      )
+    }
+
+    const saveButton = (
+      <button className={`btn btn-primary ${isValid ? '' : 'disabled'}`} onClick={onSaveClicked}>
+        <span>
+          <span className="fa fa-check-circle-o"/> Save Changes
+        </span>
+      </button>
+    );
+
+    const savingButton = (
+      <button className="btn btn-primary" disabled={true}>
+        <span className="pulsing">
+            <span className="fa fa-cog fa-spin"/> Saving&hellip;
+          </span>
+      </button>
+    );
+
+    return (
+      <div className="row footer">
+        <div className="col-md-3">
+          <button className="btn btn-default" onClick={onRevertClicked} style={{visibility: 'visible'}}>
+            <span className="glyphicon glyphicon-flash"/> Revert
```

Comment:
`fa fa-flash` instead?

Reply:
I copied this from the existing angular implementation